p2p/conn/connection_bench_test.go

```diff
@@ -566,3 +608,114 @@ func BenchmarkMConnection_Multiple_ChannelID(b *testing.B) {
 		})
 	}
 }
+
+// BenchmarkMConnection_Multiple_ChannelID assesses the max bw/send rate
+// utilization of MConnection when configured with multiple channel IDs.
+func TestMConnection_Message_Order_ChannelID(t *testing.T) {
+	// This test involves two connections, each with two channels:
+	// channel ID 1 (high priority) and channel ID 2 (low priority).
+	// It sends 11 messages from client to server, with the first 10 on
+	// channel ID 2 and the final one on channel ID 1.
+	// The aim is to show that message order at the receiver is based solely
+	// on the send order, as the receiver does not prioritize channels.
+	// To enforce a specific send order,
+	// channel ID 2's send queue capacity is limited to 1;
+	// preventing message queuing on channel ID 2 that could otherwise give
+	// priority to channel ID 1's message (despite being the last message),
+	// disrupting the send order.
+	totalMsgs := 11
+	msgSize := 1 * kibibyte
+	sendRate := 50 * kibibyte
+	recRate := 50 * kibibyte
+	clientChDesc := []*ChannelDescriptor{
+		{ID: 0x01, Priority: 1, SendQueueCapacity: 10,
+			RecvMessageCapacity: defaultRecvMessageCapacity,
+			RecvBufferCapacity:  defaultRecvBufferCapacity},
+		{ID: 0x02, Priority: 2, SendQueueCapacity: 1,
+			// channel ID 2's send queue capacity is limited to 1;
+			// to enforce a specific send order.
```

Comment:
[uber nit] I think it's slightly easier to understand comments when they are located immediately above the line they describe

```go
{
	ID: 0x02, 
	Priority: 2, 
	// channel ID 2's send queue capacity is limited to 1;
	// to enforce a specific send order.
	SendQueueCapacity: 1, 
```

or on the same line

```go
{
	ID: 0x02, 
	Priority: 2, 
	SendQueueCapacity: 1, // limited to 1 to enforce a specific send order
```

Reply:
Sure, resolved https://github.com/celestiaorg/celestia-core/pull/1193/commits/a4b979aaaf31cd6f3b93eb081075e195a630ac2f